### ISSUE TITLE
Customizing Icons for IntelliSense Completion Items

### ISSUE THREAD

**Issue by morgilad** [microsoft/vscode#223340]
#### Description

Currently, IntelliSense provides a default set of icons for completion items in the editor. However, there is no option to customize these icons based on specific needs or preferences. 

#### Proposed Solution

Introduce a feature that allows developers to customize the icons for IntelliSense completion items. This customization can be based on the kind of completion item as it is today, but with the ability to specify a different icon font.

**Comment by jrieken** [comment]
> However, there is no option to customize these icons based on specific needs or preferences.

I think this is not true. @aeschli knows for sure but AFAIK this can be customized with a contributed icon font and the only missing piece is to expose that in the monaco editor world

**Comment by aeschli** [comment]
Yes, the [product icon themes](https://code.visualstudio.com/docs/getstarted/themes#_product-icon-themes) can also change the icons of the code completion items. See [here](https://code.visualstudio.com/api/extension-guides/product-icon-theme) on how to create a product icon theme. 
The icons for code completion have ids that start with `symbol`: `symbol-array`, `symbol-boolean`, ....

**Comment by isidorn** [comment]
@morgilad I just checked and ProductIcon themes are not supported by the MonacoEditor. So the best thing you can do is overwrite the css where we define the icons for the suggest widget. Make sure that your css rules get priority, and then you can define icons for each element.

But make sure that the current icons are contributing to the monaco editor brand. And users might not like to actually see brand new icons.

![Image](https://github.com/user-attachments/assets/d9895de5-4fe9-4ad2-bd01-b1747f04f00d)

**Comment by aeschli** [comment]
For VS Code use the product icon themes.
For Monaco, define your own CSS (see https://github.com/microsoft/vscode/issues/223340#issuecomment-2374124090)
